iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -186,10 +186,13 @@ inline bool vector<T, Capacity>::emplace(const uint64_t position, Targs&&... arg
     {
         return emplace_back(std::forward<Targs>(args)...);
     }
-
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
     emplace_back(std::move(*reinterpret_cast<T*>(m_data[m_size - 1U])));
     for (uint64_t i = m_size - 1U; i > position; --i)
     {
+        // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+        // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
         reinterpret_cast<T*>(m_data)[i] = std::move(reinterpret_cast<T*>(m_data)[i - 1U]);
```

Comment:
@mossmaurice @FerdinandSpitzschnueffler @elBoberido 

Isn't this line broken? Here we just move the pointer `T*` and not the underlying object. I think the dereferencing is missing here and the correct line would be:
```cxx
*reinterpret_cast<T*>(m_data)[i] = std::move(*reinterpret_cast<T*>(m_data)[i - 1U]);
```

Do we have tests for it? Or am I blind or something ...

Reply:
@elBoberido exactly this is the problem, this line moves a pointer not a reference!

What in essence is written here is `ptr = std::move(anotherPointer);` and this is wrong, we would like to move the content of the type and use the types move assignment operator to create space for the element which is created in the new position. So we would have to write `*ptr = std::move(*anotherPointer);` since we would like to move the type! Take a look at line 191, there it is done correctly.

Reply:
Alright @mossmaurice to sum it up @elBoberido is right again but the problem was just that the brackets were set weirdly. Here is the thing:
```cxx
reinterpret_cast<T*>(m_data)[i]; // this is a T&
reinterpret_cast<T*>(m_data[i]); // this is a T* and requires an additional * to dereference it ... do you see the difference
```

Reply:
@elfenpiff Thanks for the summary. I replaced it with `at_unchecked`.